CMakeLists.txt

```diff
@@ -17,58 +17,127 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rclcpp REQUIRED)
-find_package(Boost REQUIRED)
+find_package(Boost REQUIRED COMPONENTS headers)
 
 ##############################################################################
 # Build
 ##############################################################################
 
-add_library(real_time_circular_buffer INTERFACE)
-target_include_directories(real_time_circular_buffer INTERFACE
-  include
-  ${Boost_INCLUDE_DIRS})
+add_library(realtime_circular_buffer
+  INTERFACE
+    include/filters/realtime_circular_buffer.h
+    include/filters/realtime_circular_buffer.hpp
+)
+target_link_libraries(realtime_circular_buffer
+  INTERFACE
+    Boost::boost
+)
+target_include_directories(realtime_circular_buffer
+  INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
+)
+
+add_library(filter_base
+  INTERFACE
+    include/filters/filter_base.h
+    include/filters/filter_base.hpp
+)
+target_link_libraries(filter_base
+  INTERFACE
+    rclcpp::rclcpp
+)
+target_include_directories(filter_base
+  INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
+)
+
+add_library(filter_chain
+  INTERFACE
+    include/filters/filter_chain.h
+    include/filters/filter_chain.hpp
+)
+target_link_libraries(filter_chain
+  INTERFACE
+    filter_base
+    pluginlib::pluginlib
+)
+target_include_directories(filter_chain
+  INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
+)
 
 # Plugins
 add_library(mean SHARED src/mean.cpp)
 target_link_libraries(mean
-  real_time_circular_buffer)
-ament_target_dependencies(mean
-  "pluginlib"
-  "rclcpp"
+  realtime_circular_buffer
```

Comment:
Can you specify PUBLIC or PRIVATE for the linkage type?

Reply:
address [here](https://github.com/ros/filters/pull/70/commits/6e1a2c5f7358ea680a598175a4c437ef19d5e7c5)